db/compaction/compaction.h

```diff
@@ -388,6 +389,10 @@ class Compaction {
   // should it split the output file using the compact cursor?
   InternalKey output_split_key_;
 
+  // If true, all levels in the compaction have non-overlapping files and are
+  // sorted by keys.
+  bool all_level_non_overlapping_;
```

Comment:
should it be `is_l0_trivial_move_`? Or `is_level_compaction_l0_trivial_move_`.

Reply:
I can rename it, but I assume other levels are always non overlapping, so if L0 isn't overlapping, all levels are not overlapping. I just feel odd having L0 here, as Compaction doesn't necessarily contains data from L0. Maybe it doesn't matter and I'll rename it to "l0_file_not_overlapping" if it is less confusion. I'm not sure it is always trivial move as the variable is used to determine whether it should be a trivial move or not.